code/game/machinery/nebula_shielding.dm

```diff
@@ -0,0 +1,130 @@
+///Machinery that block nebulas. This type is for convenience, you can set nebula shielding on other objects as well using add_to_nebula_shielding()
+/obj/machinery/nebula_shielding
+	density = TRUE
+
+	icon = 'icons/obj/machines/nebula_shielding.dmi'
+	pixel_x = -16
+
+	layer = FLY_LAYER
+	plane = ABOVE_GAME_PLANE
+
+	///Strength of the shield we apply
+	var/shielding_strength
+	///The type of nebula that we shield against
+	var/nebula_type
+	///How much power we use every time we block the nebula's effects
+	var/power_use_per_block = BASE_MACHINE_ACTIVE_CONSUMPTION * 2
+	///State we use when actively blocking a nebula
+	var/active_icon_state
+
+/obj/machinery/nebula_shielding/Initialize(mapload)
+	. = ..()
+
+	add_to_nebula_shielding(src, nebula_type, PROC_REF(get_nebula_shielding))
+
+///Nebula is asking us how strong we are. Return our shield strength is all is well
+/obj/machinery/nebula_shielding/proc/get_nebula_shielding()
+	SIGNAL_HANDLER
```

Comment:
~~Signal procs shouldn't be returning anything but bitflags~~ 
Just kidding but you still forgot to remove this tag.

Reply:
I'm going insane, I KNOW I removed it, urgghhhhh